Change 500 error to 400 if invalid filter applied

landonreed opened the pull request:
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Fix #257.

landonreed commented on src/main/java/com/conveyal/datatools/editor/controllers/api/EditorController.java:
```diff
@@ -217,7 +221,7 @@ private String patchTable(Request req, Response res) {
             throw e;
         } catch (Exception e) {
-            logMessageAndHalt(req, 500, "Could not patch update table", e);
+            logMessageAndHalt(req, 400, "Could not patch update table", e);
```
I disagree, Evan. Did you see the original ticket (#257)? The UI throws up a very generic response when a 5xx error is encountered. This is OK because 500 errors should only be thrown when something very mysterious happened that's really outside the bounds of our application (e.g., networking level stuff). However, in this case, we have the exception in question, so we should pass it back to the requestor so that it gets displayed correctly. Generally, we can expect that the client request had something incorrect if this catch block is triggered.

evansiroky replied:
I did read the original ticket and am aware of what happens in terms of datatools application logic when a 5xx error is returned. However, as I noted, there are a lot of SQL Exceptions that could occur due to improper system problems that an end-user is unable to control. This could be something like the database user hasn't had permissions setup properly or the postgres database crashed and/or is currently unavailable. Those are unlikely, but still possible and therefore it'd be inappropriate to return a 4xx error because it wasn't the user's fault in those cases.

landonreed commented:
@evansiroky, I just addressed your comment. For some reason the [commit](https://github.com/ibi-group/datatools-server/commit/bccb803442f0b3eb3b7777db347cb559ebcc8f6b) is not showing up for me?

landonreed commented:
@evansiroky, I just addressed your comment. For some reason the [commit](https://github.com/ibi-group/datatools-server/commit/bccb803442f0b3eb3b7777db347cb559ebcc8f6b) is not showing up for me?

evansiroky commented on src/main/java/com/conveyal/datatools/editor/controllers/api/EditorController.java:
```diff
@@ -204,7 +205,11 @@ private String patchTable(Request req, Response res) {
-                field.setParameter(preparedStatement, oneBasedIndex, filterValues.get(i));
+                try {
+                    field.setParameter(preparedStatement, oneBasedIndex, filterValues.get(i));
+                } catch (Exception e) {
```
Can this Exception type be more specific?

landonreed replied:
I'd prefer to keep it general because if an exception that we don't anticipate is thrown, no error feedback will be returned to the user at all.

landonreed commented:
:tada: This PR is included in version 3.7.0 :tada:

The release is available on [GitHub release](https://github.com/ibi-group/datatools-server/releases/tag/v3.7.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
